tests/test_self_update.cpp

```
@@ -226,6 +233,102 @@ TEST_F(InstallStdlibTest, OldLayoutArchiveInstallsToLibStd) {
+TEST_F(InstallNativeLibsTest, SkipsSubdirectories) {
+    write_file(src_dir / "lib" / "libry_parser.dylib", "parser lib");
+    write_file(src_dir / "lib" / "subdir" / "libry_nested.so", "nested lib");
+
+    bool ok = install_native_libs(src_dir.string());
+    ASSERT_TRUE(ok);
+
+    EXPECT_TRUE(fs::exists(dest_home / "lib" / "libry_parser.dylib"));
+    EXPECT_FALSE(fs::exists(dest_home / "lib" / "subdir"));
+}
+
+TEST_F(InstallNativeLibsTest, ReturnsFalseWhenDestLibPathIsAFile) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)